File: Osnovne skole/2026/filteri/solutions/main.py
import sys

input = sys.stdin.readline

def prefix_sum_2d(mat, H, W):
    # Row-wise prefix sums
    for i in range(H + 1):
        row = mat[i]
        s = 0
        for j in range(W + 1):
            s += row[j]
            row[j] = s
    # Column-wise prefix sums
    for j in range(W + 1):
        s = 0
        for i in range(H + 1):
            s += mat[i][j]
            mat[i][j] = s

def main():
    H, W = map(int, input().split())

    filteri_A = [[0] * (W + 2) for _ in range(H + 2)]
    filteri_B = [[0] * (W + 2) for _ in range(H + 2)]

    A = int(input())
    for _ in range(A):
        r1, c1, r2, c2 = map(int, input().split())
        filteri_A[r1][c1] += 1
        filteri_A[r2 + 1][c1] -= 1
        filteri_A[r1][c2 + 1] -= 1
        filteri_A[r2 + 1][c2 + 1] += 1

    B = int(input())
    for _ in range(B):
        r1, c1, r2, c2 = map(int, input().split())
        filteri_B[r1][c1] += 1
        filteri_B[r2 + 1][c1] -= 1
        filteri_B[r1][c2 + 1] -= 1
        filteri_B[r2 + 1][c2 + 1] += 1

    prefix_sum_2d(filteri_A, H, W)
    prefix_sum_2d(filteri_B, H, W)

    out = []
    for i in range(H):
        row_a = filteri_A[i]
        row_b = filteri_B[i]
        row = []
        for j in range(W):
            a = row_a[j]
            b = row_b[j]
            if a > 0 and b > 0:
                row.append('#')
            elif a > 0 or b > 0:
                row.append('/')
            else:
                row.append('.')
        out.append(''.join(row))
    sys.stdout.write('\n'.join(out) + '\n')

main()

File: Osnovne skole/2026/filteri/solutions/main.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
  int H, W, A, B;

  cin >> H >> W;

  vector <vector<int>> filteri_A(H + 2, vector<int>(W + 2)), filteri_B(H + 2, vector<int>(W + 2));

  cin >> A;

  int r1, c1, r2, c2;

  for (int i = 0; i < A; i++) {
    cin >> r1 >> c1 >> r2 >> c2;

    filteri_A[r1][c1]++;
    filteri_A[r2 + 1][c1]--;
    filteri_A[r1][c2 + 1]--;
    filteri_A[r2 + 1][c2 + 1]++;
  }

  cin >> B;

  for (int i = 0; i < B; i++) {
    cin >> r1 >> c1 >> r2 >> c2;

    filteri_B[r1][c1]++;
    filteri_B[r2 + 1][c1]--;
    filteri_B[r1][c2 + 1]--;
    filteri_B[r2 + 1][c2 + 1]++;
  }


  for (int i = 0; i < H; i++) {
    for (int j = 0; j < W; j++) {
      if (i > 0) {
        filteri_A[i][j] += filteri_A[i - 1][j];
        filteri_B[i][j] += filteri_B[i - 1][j];
      }
      if (j > 0) {
        filteri_A[i][j] += filteri_A[i][j - 1];
        filteri_B[i][j] += filteri_B[i][j - 1];
      }
      if (i > 0 && j > 0) {
        filteri_A[i][j] -= filteri_A[i - 1][j - 1];
        filteri_B[i][j] -= filteri_B[i - 1][j - 1];
      }
    }
  }

  for(int i = 0; i < H; i++) {
    for(int j = 0; j < W; j++) {
      if(filteri_A[i][j] > 0 && filteri_B[i][j] > 0) {
        cout << "#";
      } else if (filteri_A[i][j] > 0 || filteri_B[i][j] > 0) {
        cout << "/";
      } else {
        cout << ".";
      }
    }
    cout << endl;
  }


  return 0;
}
